msprime/demography.py

```diff
@@ -122,9 +122,36 @@ def insert_populations(self, tables):
         Insert population definitions for this demography into the specified
         set of tables.
         """
+        metadata_schema = tskit.MetadataSchema(
+            {
+                "codec": "json",
+                "type": "object",
+                "properties": {
+                    "name": {"type": "string"},
+                    "description": {"type": ["string", "null"]},
+                },
+                "required": ["name", "description"],
```

Comment:
Given the discussion around standardised schemas being flexible (and we intend this schema to be replaced with a reference to the standard one) - what's the thinking around `description` being required. It could be absent instead of `None`? We don't want all standardised keys to be required, so why this one?

Reply:
Yes, I agree. Initially I thought it would be nice if users of msprime could be sure that the ``description`` key was always present, even if equal to None. But, we can do that anyway and don't need to state it in the schema. So, I guess there's no real point in making ``name`` required here either, come to think of it.

Reply:
Having thought about this further - any metadata that we feel strongly enough that it be `required` in the standard schema probably should be in the data model and not metadata, right?

Reply:
Hah, actually no, I've had a (re)change of heart there. By making these fields required in the schema, we're passing on valuable information to downstream software: you're guaranteed that these values are defined in each bit of metadata. This isn't a general thing, just that in msprime's case we will *always* have these fields. The "standard" base schemas shouldn't make anything mandatory, but programs can choose to notify client code that the fields will always be present, if they wish.

Reply:
Gotcha - msprime will modify the base schema to indicate what it always provides.